addons/mail/models/discuss/discuss_channel_member.py

```diff
@@ -172,6 +173,11 @@ def create(self, vals_list):
         # help the ORM to detect changes
         res.partner_id.invalidate_recordset(["channel_ids"])
         res.guest_id.invalidate_recordset(["channel_ids"])
+        # Always link members to parent channels as well. Member list should be
+        # kept in sync.
+        for member in res:
+            if parent := member.channel_id.parent_channel_id:
+                parent.add_members(partner_ids=member.partner_id.ids, guest_ids=member.guest_id.ids)
```

Comment:
Isn't there a way we can avoid doing that?
If sub channels have their own ACL + inheriting those from parent, this might not be necessary?
What issue is this fixing?